tests/input/test_input_template.py

```diff
@@ -139,3 +145,77 @@ async def test_key_modification_actions():
         assert input.value == "     -    J-"
         input.clear()
         assert input.value == ""
+
+
+@pytest.mark.xfail(
+    reason="Cursor position is not updated correctly when the cursor is after the last separator."
+)
+async def test_cursor_word_right_after_last_separator():
+    app = InputApp(">NNN-NNN-NNN-NNNNN;_")
+    async with app.run_test():
+        input = app.query_one(Input)
+        input.value = "123-456-789-012"
+        input.cursor_position = 13
+        input.action_cursor_right_word()
+        assert input.cursor_position == 17
```

Comment:
@amottola I noticed the "cursor word right" action isn't working when the cursor is beyond the last separator.

Reply:
Hi and thanks for your review! I've already fixed the issue you mention about the "cursor word right", but I also had to modify the test: when user is on a "block" of non-separator characters, skipping to next word should bring to the beginning of the next block. Since we already are on the last block, the cursor should be moved to the last character in Input value, so the assert on line 160 should read `assert input.cursor_position == 15`. Remember the missing final two characters were not yet added by the user, so they are just visual placeholders.

Reply:
Ah, yeah, makes sense - I counted the wrong characters 😄